src/upstream_patches_ui.sh

```diff
@@ -115,8 +113,78 @@ function dashboard_entry_menu()
 # developer. For this reason, we just need to display the save patches.
 function show_bookmarked_patches()
 {
-  # TODO: Get bookmarked_patches from the local database
-  list_patches 'Bookmarked patches' bookmarked_patches
+  if [[ -z "${screen_sequence['RETURNING']}" ]]; then
+    get_bookmarked_series bookmarked_series
+  fi
+
+  # Avoiding stale value
+  screen_sequence['RETURNING']=''
+
+  list_patches 'Bookmarked patches' bookmarked_series "${screen_sequence['SHOW_SCREEN']}"
+}
+
+# Screen resposible for showing a specific bookmarked series details
+#
+# @series_index: Series index
+function show_bookmarked_series_details()
+{
+  local series_index="$1"
+  local -a action_list
+  local patch_metadata
+  local target_patch
+  local message_box
+  local columns
+  local patch_title
+  local download_dir_path
+
+  action_list=('Unbookmark')
+
+  # The local bookmark database starting index is 1 and the index
+  # passed as argument starts at 0.
+  if [[ ! "${series_index}" =~ ^[0-9]+$ || "${series_index}" -lt 0 ]]; then
+    complain "Invalid value for series_index: '${series_index}'"
+    handle_exit 22 # EINVAL
+  fi
+
+  ((series_index++))
+
+  target_patch=$(get_bookmarked_series_by_index "${series_index}")
+  IFS="${SEPARATOR_CHAR}" read -r -a columns <<< "$target_patch"
+
+  patch_metadata=$(prettify_string 'Series:' "${columns[6]}")
+  patch_metadata+=$(prettify_string 'Author:' "${columns[2]}")
+  patch_metadata+=$(prettify_string 'Version:' "${columns[4]}")
+  patch_metadata+=$(prettify_string 'Patches:' "${columns[5]}")
+
+  patch_title="${columns[6]}"
+  download_dir_path="${columns[8]}"
+
+  message_box="$patch_metadata"
+
+  create_simple_checklist 'Bookmarked Series info and actions' "$message_box" 'action_list' 1
+  ret="$?"
+
+  case "$ret" in
+    0) # OK
+      IFS=' ' read -ra selected_options <<< "$menu_return_string"
+      for option in "${selected_options[@]}"; do
+        case "$option" in
+          'Unbookmark')
+            delete_series_from_local_storage "${download_dir_path}" "${patch_title}"
+            remove_series_from_bookmark_by_index "${series_index}"
```

Comment:
Correct me if I'm wrong, but these functions can fail. If so, could you please check the return value? I think we need to notify users in case of an error.